src/components/ui/Suggest.jsx

```diff
@@ -63,78 +66,80 @@ const Suggest = ({
   );
 
   useEffect(() => {
-    const handleFocus = () => {
-      if (inputNode.value === '' || isMobile) {
-        setIsOpen(true);
-        fetchItems(inputNode.value);
-      }
-    };
-
-    const handleBlur = () => {
+    if (!hasFocus) {
       close();
-    };
+    } else if (isMobile) {
+      setIsOpen(true);
+    }
+  }, [hasFocus, isMobile]);
 
-    const handleInput = e => {
-      const { value } = e.target;
+  useEffect(() => {
+    setHighlighted(null);
+    if (value !== null) {
       fetchItems(value);
       setIsOpen(true);
-      setLastQuery(value);
-    };
+    }
+  }, [fetchItems, value]);
```

Comment:
These 2 `useEffect` do not handle correctly the case when a empty input is focused. A list of favorites should appear as suggestions and it's no longer the case (probably related to the condition `inputNode.value === '' ` that has been removed).

Reply:
I replaced these 2 effects by a single one, which makes things actually simpler. I hope I haven't overlooked yet another edge-case…